packages/grafana-data/src/transformations/transformDataFrame.ts

```diff
@@ -1,37 +1,68 @@
+import { MonoTypeOperatorFunction, Observable, of, OperatorFunction } from 'rxjs';
+import { map, mergeMap } from 'rxjs/operators';
+
 import { DataFrame, DataTransformerConfig } from '../types';
-import { standardTransformersRegistry } from './standardTransformersRegistry';
+import { standardTransformersRegistry, TransformerRegistyItem } from './standardTransformersRegistry';
 
-/**
- * Apply configured transformations to the input data
- */
-export function transformDataFrame(options: DataTransformerConfig[], data: DataFrame[]): DataFrame[] {
-  let processed = data;
-  for (const config of options) {
-    const info = standardTransformersRegistry.get(config.id);
-
-    if (!info) {
-      return data;
-    }
-
-    const defaultOptions = info.transformation.defaultOptions ?? {};
-    const options = { ...defaultOptions, ...config.options };
-    const transformer = info.transformation.transformer(options);
-    const after = transformer(processed);
-
-    // Add a key to the metadata if the data changed
-    if (after && after !== processed) {
+const getOperator = (config: DataTransformerConfig): MonoTypeOperatorFunction<DataFrame[]> => source => {
+  const info = standardTransformersRegistry.get(config.id);
+
+  if (!info) {
+    return source;
+  }
+
+  const defaultOptions = info.transformation.defaultOptions ?? {};
+  const options = { ...defaultOptions, ...config.options };
+
+  return source.pipe(
+    mergeMap(before => of(before).pipe(info.transformation.operator(options), postProcessTransform(before, info)))
+  );
+};
```

Comment:
```suggestion
const getOperator = (config: DataTransformerConfig): MonoTypeOperatorFunction<DataFrame[]> => {
  const info = standardTransformersRegistry.get(config.id);

  if (!info) {
    return pipe();
  }

  const defaultOptions = info.transformation.defaultOptions ?? {};
  const options = { ...defaultOptions, ...config.options };
  const operator = info.transformation.operator(options);
  
  return pipe(mergeMap(before => of(before).pipe(operator,  postProcessTransform(before, info))));
};
```

nit: I'm not 100% sure that you can return an empty pipe but might be worth testing to clean up the code a bit.

Reply:
Hmm I don't agree that it adds to readability and TypeScript complains about it. Maybe we can enhance this after this PR?
```ts
Type '(source: Observable<DataFrame[]>) => UnaryFunction<DataFrame[], unknown>' is not assignable to type 'MonoTypeOperatorFunction<DataFrame[]>'.
  Type 'UnaryFunction<DataFrame[], unknown>' is missing the following properties from type 'Observable<DataFrame[]>': _isScalar, source, operator, lift, and 6 more.
```